### ISSUE TITLE
Doesn't detect inotify events on /proc/selfpid/mem

### ISSUE THREAD

**Issue by kamikazechaser** [fsnotify/fsnotify#658]
Generally, a program has the ability to "add watch" to inotify to listen for events e.g. IN_OPEN on `/proc/selfpid/mem` to monitor any access to its memory. The general use case here is to monitor for any memory dumping or overwrites e.t.c.

```go
package main

import (
	"fmt"
	"log"
	"os"

	"github.com/fsnotify/fsnotify"
)

func main() {
	watcher, err := fsnotify.NewWatcher()
	if err != nil {
		log.Fatal(err)
	}
	defer watcher.Close()

	go func() {
		for {
			select {
			case event, ok := <-watcher.Events:
				if !ok {
					return
				}
				log.Println("event:", event)
			case err, ok := <-watcher.Errors:
				if !ok {
					return
				}
				log.Println("error:", err)
			}
		}
	}()

	memoryFile := fmt.Sprintf("/proc/%d/mem", os.Getpid())
	log.Println("memory file:", memoryFile)
	err = watcher.Add(memoryFile)

	if err != nil {
		log.Fatal(err)
	}

	<-make(chan struct{})
}
```

I would expect the above code to log an OPEN event if any attempt was made to read the memory even when using a simple command like `cat proc/pid/mem`. However, nothing is logged.

When I switch to `golang.org/x/sys/unix`, I can receive the raw inotify events.

Is there any reason why these events aren't being picked up by the library?

**Comment by xplshn** [comment]
Doesn't work for any file in procfs, I tried with /proc/acpi/button/lid/LID0/state

**Comment by arp242** [comment]
Yes, this is a documented limitation: https://github.com/fsnotify/fsnotify?tab=readme-ov-file#why-dont-notifications-work-with-nfs-smb-fuse-proc-or-sys

Not much we can do about it; the Linux kernel/inotify doesn't send events.
